sys/net/rtsock.c

```diff
@@ -1255,21 +1256,15 @@ rtsock_msg_mbuf(int type, struct rt_addrinfo *rtinfo)
 		}
 #endif
 		m_copyback(m, len, sa->sa_len, (caddr_t)sa);
+		/*
+		 * XXX-AM: Assume that SA_SIZE aligns to a multiple of sizeof(long),
+		 * so at most we have to copy sizeof(long) bytes.
+		 */
 		if (dlen - sa->sa_len > 0) {
-			/*
-			 * Add any padding required by SA_SIZE on the receiver
-			 * side.
-			 * Assume that mbuf is always allocated as a multiple
-			 * of the alignment enforced by SA_SIZE().
-			 * XXX-AM: This is a bit gross but avoids calling
-			 * m_copyback or m_append to copy zeroes.
-			 */
-			KASSERT(M_TRAILINGSPACE(m) >= dlen - sa->sa_len,
-			    ("No space for padding in rtsock mbuf"));
-			bzero(mtod(m, caddr_t) + len + sa->sa_len,
-			    dlen - sa->sa_len);
-			if (m->m_flags & M_PKTHDR)
-				m->m_pkthdr.len += dlen - sa->sa_len;
+			KASSERT(dlen - sa->sa_len <= sizeof(long),
```

Comment:
```suggestion
			char zero_padding[sizeof(long)] = {0};

			KASSERT(dlen - sa->sa_len <= sizeof(long),
```
A zero-filled char array would avoid the need for a cast. Don't really mind either way.